tesseract/messaging/evm/src/gas_oracle.rs

```diff
@@ -197,43 +207,41 @@ pub async fn get_current_gas_cost_in_usd(
 					gas_price_cost = convert_27_decimals_to_18_decimals(unit_wei * gas_price)?;
 				},
 				POLYGON_CHAIN_ID | POLYGON_TESTNET_CHAIN_ID => {
-					let uri = format!(
-					"{api}?chainid={POLYGON_CHAIN_ID}&module=gastracker&action=gasoracle&apikey={api_keys}"
-				);
 					let node_gas_price = client.get_gas_price().await?;
-					let response_json = make_request::<GasResponse>(&uri, Default::default())
-						.await
-						.unwrap_or_default();
-					let oracle_gas_price =
-						parse_units(response_json.result.safe_gas_price, "gwei")?.into();
-					// needed because of ether-rs and polkadot-sdk incompatibility
-					gas_price = if inner_evm == POLYGON_CHAIN_ID {
-						new_u256(std::cmp::max(node_gas_price, oracle_gas_price))
-					} else {
-						new_u256(node_gas_price)
-					};
-					let eth_usd = parse_to_27_decimals(&response_json.result.usd_price)?;
-					unit_wei = get_cost_of_one_wei(eth_usd);
-					gas_price_cost = convert_27_decimals_to_18_decimals(unit_wei * gas_price)?;
+					gas_price = new_u256(node_gas_price);
+
+					let native_token = "0x0d500B1d8E8eF31E21C99d1Db9A6444d3ADf1270".parse::<H160>()?;
```

Comment:
Instead of hardcoding these addresses and fee token, Import the full uniswap v2 abi file, 
Fetch the native token address from the uniswap contract by calling WETH()
Fetch the fee token address from the ismp host params, 
Fetch the fee token  decimals from the erc20 interface